packages/realm/react-native/ios/RealmReact/RealmReact.mm

```diff
@@ -0,0 +1,184 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RealmReact.h"
+
+#import <realm-js-ios/jsi_init.h>
+
+#import <React/RCTBridge+Private.h>
+#import <React/RCTInvalidating.h>
+#import <ReactCommon/CallInvoker.h>
+#import <jsi/jsi.h>
+
+#import <arpa/inet.h>
+#import <ifaddrs.h>
+#import <net/if.h>
+#import <netdb.h>
+#import <objc/runtime.h>
+
+// the part of the RCTCxxBridge private class we care about
+@interface RCTBridge (Realm_RCTCxxBridge)
+- (void *)runtime;
+// Expose the CallInvoker so that we can call `invokeAsync`
+- (std::shared_ptr<facebook::react::CallInvoker>)jsCallInvoker;
+@end
+
+@interface RealmReact () <RCTBridgeModule, RCTInvalidating>
+@end
+
+@implementation RealmReact {
+  NSMutableDictionary *_eventHandlers;
+  // Keep track of whether we are already waiting for the React Native UI queue
+  // to be flushed asynchronously
+  bool waitingForUiFlush;
+}
+
+@synthesize bridge = _bridge;
+
+RCT_EXPORT_MODULE(Realm)
+
++ (BOOL)requiresMainQueueSetup {
+  return YES;
+}
+
++ (void)initialize {
+  if (self != [RealmReact class]) {
+    return;
+  }
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _eventHandlers = [[NSMutableDictionary alloc] init];
+    waitingForUiFlush = false;
+  }
+  return self;
+}
+
+- (dispatch_queue_t)methodQueue {
+  return dispatch_get_main_queue();
+}
+
+- (NSDictionary *)constantsToExport {
+  return @{};
+}
+
+- (void)addListenerForEvent:(NSString *)eventName
+                    handler:(RealmReactEventHandler)handler {
+  NSMutableOrderedSet *handlers = _eventHandlers[eventName];
+  if (!handlers) {
+    handlers = _eventHandlers[eventName] = [[NSMutableOrderedSet alloc] init];
+  }
+  [handlers addObject:handler];
+}
+
+- (void)removeListenerForEvent:(NSString *)eventName
+                       handler:(RealmReactEventHandler)handler {
+  NSMutableOrderedSet *handlers = _eventHandlers[eventName];
+  [handlers removeObject:handler];
+}
+
+RCT_REMAP_METHOD(emit, emitEvent
+                 : (NSString *)eventName withObject
+                 : (id)object) {
+  for (RealmReactEventHandler handler in [_eventHandlers[eventName] copy]) {
+    handler(object);
+  }
+}
```

Comment:
I actually don't think this is used / needed anymore 🤔

Reply:
Just saw this comment.  They can be removed.  These were used by the legacy react native test project.

Reply:
I removed it ..